### ISSUE TITLE
restrict_kubeapi not working

### ISSUE THREAD

**Issue by Nils98Ar** [SovereignCloudStack/k8s-cluster-api-provider#630]
I've set this in the tf environment:
```
[...]
restrict_kubeapi = [ "none" ]
[...]
```

This creates the following file in the workload cluster dir:
```
ubuntu@pub-01-mgmtcluster:~/pub-dev-01$ cat restrict-kubeapi-cidr.yaml
---
apiVersion: infrastructure.cluster.x-k8s.io/v1alpha7
kind: OpenStackCluster
metadata:
  name: pub-dev-01
spec:
  allowAllInClusterTraffic: true
  apiServerLoadBalancer:
    allowedCidrs:
    - <external ip>/32
```
But it is not patched into the actual workload cluster config yaml:
```
ubuntu@pub-01-mgmtcluster:~/pub-dev-01$ grep -i cidr pub-dev-01-config.yaml
      cidrBlocks:
      cidrBlocks:
  nodeCidr: <xyz>
```
.. and therefore not to applied the OpenStack LB:

```
ubuntu@pub-01-mgmtcluster:~/pub-dev-01$ openstack loadbalancer listener show k8s-clusterapi-cluster-pub-dev-01-pub-dev-01-kubeapi-6443 | grep -i cidr
| allowed_cidrs               | None
```

**Comment by Nils98Ar** [comment]
Adding the namespace already fixes it for me but this might not be backwards compatible for older versions without cluster namespaces.
```
kind: OpenStackCluster
metadata:
  name: ${CLUSTER_NAME}
  namespace: ${CLUSTER_NAME}
spec:
```

**Comment by chess-knight** [comment]
This is exactly what I discovered yesterday and already proposed a fix in #625.

**Comment by Nils98Ar** [comment]
Ahh I have only searched the issues :)
